### DISCUSSION TITLE
Ability to code without using semicolon

### DISCUSSION THREAD

**Discussion by RobFosterNYC** [dotnet/csharplang#5620] [Language Ideas] [answered]
Would love to have the same ability to code without using semicolon.
Several languages already do this.  By giving the developer the option to code both ways, either with semicolon or without.

Python and Javascript does this.
I am hoping that it can also be done in C#

thanks.

**Comment by HaloFour** [comment, marked as answer]
#1581
#496

https://mobile.twitter.com/jaredpar/status/1010195445887336450

**Comment by DavidArno** [comment]
> I am hoping that it can also be done in C#

I suspect that the only way you'll avoid that hope turning to disappointment would be to buy Microsoft, sack the existing C# language team and install a new team that will carry out your wishes. 😱

**Reply by orthoxerox** [reply]
I think that's a bit excessive. You can simply fork dotnet/roslyn, rewrite the grammar to exclude semicolons, work out all the edge cases, fix all the tests, release your own dialect of C# and start promoting it over MS C#.

**Reply by DavidArno** [reply]
To be honest, my approach sounds like a lot less work 🤣

**Reply by RobFosterNYC** [reply]
> I think that's a bit excessive. You can simply fork dotnet/roslyn, rewrite the grammar to exclude semicolons, work out all the edge cases, fix all the tests, release your own dialect of C# and start promoting it over MS C#.

Your comment was the MOST constructive comment I have ever come across in Github or any other platform.  Thank you for being a human-being.
Its funny how everyone is losing their shit over a semicolon comment, I made.
And the thing is, I never said to those people or anyone for that matter, to get rid of it entirely, I just said give the option not to use it.
The operative word that everyone seems to have passed over is  "Option".

As I mentioned before Python and Javascript gives people the option not to use it and I recently found out that even F# and a few others also gives this option.
I have had people go to my social media and make vague threats or wish death on me and my family over this damn comment post.
And they know they are wrong by making a new account, just to deliberately make the threat, because they are too chicken to reveal themselves.
Why is it that if you have an opinion that differs from others, people come out the wood works with horrible intentions & threats?
I didn't take their first born or committed any harm towards them, but yet they do this onto other people.
And the thing is, they would never dare say this to peoples faces, for obvious reasons. They cowardly hide behind the internet screen.

Its like going to a star wars convention, then you say you like star trek a little more, 
next thing you know, you have a mob trying to kill you over an opinion. Why people have this type of mob mentality onto others?

This is why I hate posting on forums of any kind. Because you can't have an opinion without being ridiculed by a mob of crazy people,
who believe their way is the only way that exist, and if you try to change it in any way, shape or form, then off with your head.
Each time, this just teaches me to stay away from forums, and just email the company directly to avoid crazy people.

There is a word for this, I just cant think of it at this moment.

Anyways, thank you @orthoxerox for being a human-being.  sadly there are very few left on this planet.

**Reply by 333fred** [reply]
>  I have had people go to my social media and make vague threats or wish death on me and my family over this damn comment post.

To be clear, this is never acceptable. Besides being a violation of our CoC, it's just a horrible thing to do. If you have received such threats, please reach out to conduct@dotnetfoundation.org and the conduct team will see what they can do.

**Reply by CyrusNajmabadi** [reply]
> Its funny how everyone is losing their shit over a semicolon comment, I made.

I'm a little bit confused by this.  I see 5 responses to your post.  All of which are constructive and on topic.  No one here appears to be taking issue with this. 

> I have had people go to my social media and make vague threats or wish death on me and my family over this damn comment post.

Could you please link us to this (feel free to use conduct@dotnetfoundation.org) so we can take a look?  Thanks!

**Reply by TahirAhmadov** [reply]
Actually both JS and Python suffer from lack of `;` enforcement. In Python, they solve it with whitespace enforcement; that's a significantly worse solution. Ask any Python developer who had fails because of tab vs space character like situations. In JS, there are other problems - for example, you cannot put the expression on a different line than the `return` keyword, or else it will return `undefined` and the expression becomes a separate statement, which is usually in effect a no-op, which is also never executed (because it's right after a `return`). In fact, lack of semicolons is one of the reasons why I refuse to use Python for anything; and whenever I code in JS, I follow a strict self-imposed "always use semicolon" rule.
Trust me on this - we have decades of experience as an industry on this. Semicolons are better.
Furthermore, languages which don't enforce semicolons currently, like T-SQL, should start enforcing them sooner than later. (They keep postponing it, to avoid pissing off their userbase.)

**Comment by huoyaoyuan** [comment]
I **strongly disagree** to use new line for statement separation. Having no semantic meaning for new line makes C family languages can use new line arbitrarily for formatting, without bothering about line continuation.
